litex/tools/litex_sim.py

```diff
@@ -545,7 +542,7 @@ def pre_run_callback(vns):
 
     builder = Builder(soc, **parser.builder_argdict)
     builder.build(
-        sim_config       = sim_config,
+        sim_config       = soc.sim_config,
```

Comment:
Because `sim_config` needs to be passed to the builder but is not in `main()` anymore, `SimSoC()` exposes it as `self.sim_config`.